services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteOptions.java

```diff
@@ -2019,6 +2032,23 @@ public Builder snappingIncludeClosuresList(@Nullable List<Boolean> snappingClosu
     @NonNull
     public abstract Builder experimental(@Nullable Experimental experimental);
 
+    /**
+     * Use this method to add request parameters which are not present in model yet but supported
+     * on the Directions API, for example experimental parameters.
+     * @param unrecognizedProperties parameters to add to request
+     */
+    @NonNull
+    public Builder unrecognizedProperties(Map<String, String> unrecognizedProperties) {
+      LinkedHashMap<String, SerializableJsonElement> mapped = new LinkedHashMap<>();
+      for (Map.Entry<String, String> entry : unrecognizedProperties.entrySet()) {
+        mapped.put(
+          entry.getKey(),
+          new SerializableJsonElement(new JsonPrimitive(entry.getValue()))
+        );
+      }
+      return unrecognized(mapped);
```

Comment:
```suggestion
    public Builder unrecognizedProperties(@Nullable Map<String, JsonPrimitive> unrecognizedProperties) {
      LinkedHashMap<String, SerializableJsonElement> mapped = null;
      if (unrecognizedProperties != null) {
        mapped = new LinkedHashMap<>();
        for (Map.Entry<String, String> entry : unrecognizedProperties.entrySet()) {
          mapped.put(
            entry.getKey(),
            new SerializableJsonElement(entry.getValue())
          );
        }
      }
      return unrecognized(mapped);
```

Suggesting to declare the nullability and support all `JsonPrimitives`, not only `string`s, for convenience as that's what the parser above expects.

Reply:
I made the parameter nullable